src/contracts/QBridge.h

```diff
@@ -0,0 +1,362 @@
+    id admin;                                      // Admin address
+    QPI::HashMap<id, bit, 16> managers;            // Managers list
+    uint64 totalReceivedTokens;                    // Total tokens received
+
+    // Internal methods for admin/manager permissions using qpi CALLS
+    PRIVATE_FUNCTION_WITH_LOCALS(isAdmin)
+        output = (qpi.invocator() == state.admin);
+    _
+
+        PRIVATE_FUNCTION_WITH_LOCALS(isManager)
+        output = state.managers.get(qpi.invocator(), bit());
+    _
+
+        /*
+        // Check if the invocator is the admin
+        bool isAdmin() const {
+            return qpi.invocator() == state.admin;
+        }
+
+        // Check if the invocator is a manager
+        bool isManager() const {
+            bit isManager = false;
+            state.managers.get(qpi.invocator(), isManager);
+            return isManager;
+        }*/
+
+public:
+    // Create a new order and lock tokens
+    PUBLIC_FUNCTION_WITH_LOCALS(createOrder)
+
+        // Validate the input 
+        if (input.amount == 0) {
+            output.status = 1; // Error
+            copyMemory(output.message, "Amount must be greater than 0");
+            return;
+        }
+
+        if (qpi.invocationReward() < state.transactionFee) {
+            output.status = 2; // Error
+            copyMemory(output.message, "Insufficient transaction fee");
+            return;
+        }
+
+        // Create the order
+        BridgeOrder newOrder;
+        newOrder.orderId = state.nextOrderId++;
+        newOrder.qubicSender = qpi.invocator();
+        newOrder.ethAddress = input.ethAddress;
+        newOrder.amount = input.amount;
+        newOrder.orderType = 0; // Default order type
+        newOrder.status = 0; // Created
+        newOrder.fromQubicToEthereum = input.fromQubicToEthereum;
+
+        // Store the order
+        state.orders.set(newOrder.orderId, newOrder);
+
+        output.status = 0; // Success
+        copyMemory(output.message, "Order created successfully");
+    _
+
+    // Retrieve an order
+    PUBLIC_FUNCTION_WITH_LOCALS(getOrder)
+
+        BridgeOrder order;
+        if (!state.orders.get(input.orderId, order)) {
+            output.status = 1; // Error
+            copyMemory(output.message, "Order not found");
+            return;
+        }
+
+        OrderResponse orderResp;
+        // Populate OrderResponse with BridgeOrder data
+        orderResp.orderId = order.orderId;
+        orderResp.originAccount = order.qubicSender;
+        orderResp.destinationAccount = order.ethAddress;
+        orderResp.amount = order.amount;
+        copyMemory(orderResp.memo, "Bridge transfer details"); // Placeholder for metadata
+        orderResp.sourceChain = state.sourceChain;
+
+        output.status = 0; // Success
+        output.order = orderResp;
+        copyMemory(output.message, "Order retrieved successfully");
+    _
+
+    // Admin Functions
+    PUBLIC_PROCEDURE_WITH_LOCALS(setAdmin)
```

Comment:
This requires no locals, so you can remove the `_WITH_LOCALS`. Applies to some below as well.